Makefile

```diff
@@ -2,28 +2,32 @@ VERSION := $(shell git describe --always --tags --dirty)
 ldflags := "-X sqyrrl/server.Version=${VERSION}"
 build_args := -a -v -ldflags ${ldflags}
 
-build_path = "build/sqyrrl-${VERSION}"
+build_path = build/sqyrrl-${VERSION}
 
-CGO_ENABLED := 1
 GOARCH := amd64
 
+CGO_ENABLED := 1
+
 .PHONY: build build-linux build-darwin build-windows check clean coverage install lint test
 
 all: build
 
 build: build-linux build-darwin build-windows
 
+build-linux: GOOS = linux
 build-linux:
 	mkdir -p ${build_path}
-	GOARCH=${GOARCH} GOOS=linux go build ${build_args} -o ${build_path}/sqyrrl-linux-${GOARCH} ./main.go
```

Comment:
This change loses propagation of the above `GOARCH := amd64` to the `go build` command: on arm I get a binary named "amd64" with aarch64 architecture :
```
vscode ➜ /workspaces/go/sqyrrl (c719b24) $ file build/sqyrrl-1.2.0-16-gc719b24/sqyrrl-linu*
build/sqyrrl-1.2.0-16-gc719b24/sqyrrl-linux-amd64: ELF 64-bit LSB executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, Go BuildID=nwLCKuf9DCATA-RIawA9/-pLl1pofh3Dvhs0yk2ol/1qTL1Gf4G5061T-OBFqc/SdFUOzJ99rwO2sgDjTm2, with debug_info, not stripped
```
(bug previously fixed in e2812c7)

Reply:
How is `GOARCH` being set on yout machine? Here it is empty, so the previous version had the same behaviour as the current one i.e. I had to call `make GOARCH=arm64` on Mac to override the `amd64` value.

Reply:
I think a better solution is to call `go env GOARCH` to get the correct value for the current host and then use that as the default.

Reply:
Purely from line 8 (or 7 post patch) in the Makefile.

Reply:
This [bca0270](https://github.com/wtsi-npg/sqyrrl/pull/74/commits/bca0270cde914180b323a95d5874e48eb38cb130) does fix the bug I saw (inconsistent file name and arch).